clang/test/Driver/sycl-offload-intelfpga-emu.cpp

```diff
@@ -11,7 +11,7 @@
 // RUN:  | FileCheck -check-prefixes=CHK-FPGA-LINK,CHK-FPGA-EARLY %s
 // RUN:  %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fintelfpga -fsycl-link=image %t.o -o libfoo.a 2>&1 \
 // RUN:  | FileCheck -check-prefixes=CHK-FPGA-LINK,CHK-FPGA-IMAGE %s
-// RUN:  %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice -g -fsycl-link=image %t.o -o libfoo.a 2>&1 \
+// RUN:  %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice -fsycl-link=image %t.o -o libfoo.a 2>&1 \
```

Comment:
Is it that explicit FPGA triple + explicit `-g` option usage should result in debug-aware device compilation because the user explicitly requested debug information as opposed to the `fintelfpga` case?

An additional thought that's occurred to me: should any of the compiler documentation mention the debug info nuances introduced by this PR?

Reply:
`-fintelfpga` implies `-g` under all circumstances.  My change prevents `-g` from being enabled for FPGA emulation.  Use of the triple does not imply `-g` which is why my test update was made.   To enable, user has to use `-Xsycl-target-backend --bo=-g`.  Probably not the ideal way of doing this but due to the nature of how we scan the AOT options intercepting the top level `-g` is not possible without a refactor of `AddImpliedArgs`.

Reply:
> -fintelfpga implies -g under all circumstances. My change prevents -g from being enabled for FPGA emulation.

This part is clear to me, as it appears to be the main purpose of the change.

> Use of the triple does not imply -g which is why my test update was made.

This in particular is what I've been trying to understand. Explicit use of the triple doesn't imply `-g`, therefore no exclusion of "default" `--bo=-g` appendix is needed. This in turn leads to `-fsycl-targets=spir64_fpga-unknown-unknown-sycldevice -g` auto-propagating `-g` to the AOT call, which provides on-device debug symbols but slows down the compilation. Are we OK with this scenario, given that the user has explicitly requested debug info? To my mind, the latter is decent rationale, but it would be best to have your confirmation.

> To enable, user has to use -Xsycl-target-backend --bo=-g. Probably not the ideal way of doing this but due to the nature of how we scan the AOT options intercepting the top level -g is not possible without a refactor of AddImpliedArgs.

Totally fine from my personal perspective - when skimming through the PR, I've also come to the conclusion that excessive refactoring of `AddImpliedArgs` and related functions would've been needed.

Reply:
> Are we OK with this scenario, given that the user has explicitly requested debug info? To my mind, the latter is decent rationale, but it would be best to have your confirmation.
Yes, I'm OK with this scenario.